graphqlmetrics/pkg/telemetry/metrics_prometheus.go

```diff
@@ -0,0 +1,79 @@
+package telemetry
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.uber.org/zap"
+
+	otelprom "go.opentelemetry.io/otel/exporters/prometheus"
+)
+
+func NewPrometheusServer(logger *zap.Logger, listenAddr string, path string, registry *prometheus.Registry) *http.Server {
+	r := chi.NewRouter()
+	r.Use(middleware.Recoverer)
+	r.Handle(path, promhttp.HandlerFor(registry, promhttp.HandlerOpts{
+		EnableOpenMetrics: true,
+		ErrorLog:          zap.NewStdLog(logger),
+		Registry:          registry,
+		Timeout:           10 * time.Second,
+	}))
+
+	svr := &http.Server{
+		Addr:              listenAddr,
+		ReadTimeout:       1 * time.Minute,
+		WriteTimeout:      1 * time.Minute,
+		ReadHeaderTimeout: 2 * time.Second,
+		IdleTimeout:       30 * time.Second,
+		ErrorLog:          zap.NewStdLog(logger),
+		Handler:           r,
+	}
+
+	logger.Info("Prometheus metrics enabled", zap.String("listen_addr", svr.Addr), zap.String("endpoint", path))
+
+	return svr
+}
+
+func (c *Config) NewPrometheusMeterProvider() (*sdkmetric.MeterProvider, *prometheus.Registry, error) {
+	var registry *prometheus.Registry
+
+	if c.Prometheus.TestRegistry != nil {
+		registry = c.Prometheus.TestRegistry
+	} else {
+		registry = prometheus.NewRegistry()
+	}
+
+	registry.MustRegister(collectors.NewGoCollector())
+
+	// Only available on Linux and Windows systems
+	registry.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+
+	promExporter, err := otelprom.New(
+		otelprom.WithoutUnits(),
+		otelprom.WithRegisterer(registry),
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	mp := sdkmetric.NewMeterProvider(
+		sdkmetric.WithReader(promExporter),
+		sdkmetric.WithResource(resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String(c.Name),
+		)),
+	)
+
+	otel.SetMeterProvider(mp)
```

Comment:
This will set a global meter. This can be an issue with concurrent tests. We only work with dedicated meters.